authors/apps/articles/views.py

```diff
@@ -475,3 +477,113 @@ def get(self, request, slug):
+                return Response({
+                            'error': 'The start_index_position should'
+                            ' not be greater or equal end_index_position'})
+
+        except ValueError:
+            # ensure that selection indices are integers
+            return Response({
+                'error': 'Start of highlight and end of'
+                ' highlight indices should be both integers',
+            }, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        selected_text = str(
+            single_article_instance.body[selection_range[0]:selection_range[1]])
+        comment['selected_text'] = selected_text
+
+        serializer = HighlightSerializer(data=comment, partial=True)
+        if serializer.is_valid():
+            serializer.save(user_id=request.user,
+                            article=single_article_instance)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def select_index_position(self, start_index_position, end_index_position):
+        """Stores the index values  in a variable."""
+        index_values = [start_index_position, end_index_position]
+        return index_values
+
+    def get(self, request, slug):
+        """Get all comment for highligted text of a single article."""
+        article = get_object_or_404(Article, slug=slug)
+        highlighted_comment = article.highlight.filter()
+        serializer = self.serializer_class(highlighted_comment, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class RetrieveUpdateDeleteComments(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly, )
+    serializer_class = HighlightSerializer
+    queryset = HighlightTextModel.objects.all()
+    lookup_field = 'id'
+
+    def get(self, request, slug, id):
+        article = get_object_or_404(Article, slug=slug)
+        try:
+            highlightcomment = article.highlight.filter(id=id).first().pk
+            return super().get(request, highlightcomment)
+        except:
+            message = {"error": "This comment doesnot exist"}
+            return Response(message, status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, slug, id):
+            article = get_object_or_404(Article, slug=slug)
+            try:
+                highlightcomment = article.highlight.filter(id=id).first().pk
+            except:
+                message = {"error": "The comment does not exist"}
+                return Response(message, status.HTTP_404_NOT_FOUND)
+            super().delete(self, request, highlightcomment)
+            message = {"message": "Comment on highligted text deleted successfully"}
+            return Response(
+                message, status.HTTP_200_OK)
+
+    def update(self, request, slug, id):
+        article = Article.objects.filter(slug=slug).first()
```

Comment:
Can we have a method that does the job of checking if the article and comment exist and can be called on the get, delete and update methods